includes/Media/Base_Color.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class Image_Size
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Media;
+
+use Google\Web_Stories\Service_Base;
+use Google\Web_Stories\Infrastructure\HasMeta;
+
+/**
+ * Class Base_Color
+ *
+ * @package Google\Web_Stories\Media
+ */
+class Base_Color extends Service_Base implements HasMeta {
+
+	/**
+	 * The base color meta key.
+	 *
+	 * @var string
+	 */
+	const BASE_COLOR_POST_META_KEY = 'web_stories_base_color';
+
+	/**
+	 * Init.
+	 *
+	 * @since 1.15.0
+	 *
+	 * @return void
+	 */
+	public function register() {
+		$this->register_meta();
+
+		add_filter( 'wp_prepare_attachment_for_js', [ $this, 'wp_prepare_attachment_for_js' ] );
+	}
+
+	/**
+	 * Register meta
+	 *
+	 * @since 1.15.0
+	 *
+	 * @return void
+	 */
+	public function register_meta() {
+		register_meta(
+			'post',
+			self::BASE_COLOR_POST_META_KEY,
+			[
+				'type'           => 'array',
+				'description'    => __( 'Attachment base color', 'web-stories' ),
+				'show_in_rest'   => [
+					'schema' => [
+						'type'  => 'array',
+						'items' => [
+							'type' => 'integer',
+						],
```

Comment:
Yesterday we talked about storing hex colors instead of RGB arrays, discussing advantages like easier sanitization etc.

I just learned that the REST API  (since WP 5.5) supports a special `hex-color` format.

So we could do

```suggestion
						'type'  => 'string',
						'format' => 'hex-color',
```

Just mentioning this here for context.

Reply:
Added some more php level validation.

Reply:
I know about the hex-color validation, but I don't think it makes sense to store as one value and convert it in javascript. If the javascript needs an array